### PR TITLE
ci(dev-deps): remove axios

### PR DESCRIPTION
**Pull request by mojito317**
## PR summary

The goal of this PR is to get rid of the `axios` dev dependency.
I used Node's `http` instead in the tests where wiremock was used.

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: test/integration/readme.integration.test.js [hunk 1]
```diff
@@ -40,7 +38,24 @@ describe('Readme integration tests', () => {
...
+    const p = new Promise((resolve, reject) => {
+      request
+        .on('response', (resp) => {
+          resolve(resp);
+        })
+        .on('error', (err) => {
+          reject(err);
+        });
+    });
+    await p; // Reset Wiremock states
```
Comment (ricellis) [inline test/integration/readme.integration.test.js:58]: Do you think we need any `.then()` or `catch()` here?

Actually it looks like [Jest can handle the promise](https://jestjs.io/docs/setup-teardown#one-time-setup) so maybe it makes sense to remove the `await` (and the corresponding `async` on the function) and just return the promise (either calling `setAuthentication()` beforehand or chaning it in a `then()`).

**Review by ricellis — CHANGES_REQUESTED** [review]
The test copyright needs an update and a couple of suggestions.

File: test/integration/readme.integration.test.js (hunk 1 above)
Comment (mojito317) [reply test/integration/readme.integration.test.js]: done in [50055c9](https://github.com/IBM/cloudant-node-sdk/pull/710/commits/50055c9ab06d9dd00ddcac5ac77ce6af77bb7e7e)

File: test/integration/readme.integration.test.js [hunk 2]
```diff
@@ -40,7 +38,24 @@ describe('Readme integration tests', () => {
...
+    request.end();
+    const p = new Promise((resolve, reject) => {
+      request
+        .on('response', (resp) => {
+          resolve(resp);
+        })
+        .on('error', (err) => {
+          reject(err);
+        });
+    });
```
[the remark this answers is not in the document]
Comment (mojito317) [reply test/integration/readme.integration.test.js:56]: done in [50055c9](https://github.com/IBM/cloudant-node-sdk/pull/710/commits/50055c9ab06d9dd00ddcac5ac77ce6af77bb7e7e)

File: test/integration/readme.integration.test.js [hunk 3]
```diff
@@ -40,7 +38,24 @@ describe('Readme integration tests', () => {
...
   beforeAll(async () => {
-    await axios.post(`${process.env.WIREMOCK_URL}/__admin/scenarios/reset`); // Reset Wiremock states
+    const wiremockURL = new URL(process.env.WIREMOCK_URL);
+    const request = http.request({
+      host: wiremockURL.hostname,
+      port: wiremockURL.port,
+      path: '/__admin/scenarios/reset',
+      method: 'POST',
+    });
+    request.end();
```
[the remark this answers is not in the document]
Comment (mojito317) [reply test/integration/readme.integration.test.js]: done in [50055c9](https://github.com/IBM/cloudant-node-sdk/pull/710/commits/50055c9ab06d9dd00ddcac5ac77ce6af77bb7e7e)

**Review by ricellis — APPROVED** [review]

**Review by tomblench — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @ricellis and @tomblench! I'm about to merge this!

**Pull request merged**